internal/flux/registry_mirror.go

```diff
@@ -0,0 +1,130 @@
+func GetRegistryMirrorConfig(ctx context.Context, k8sClient client.Reader, plugin *greenhousev1alpha1.Plugin) (*RegistryMirrorConfig, error) {
+	org := &greenhousev1alpha1.Organization{}
+	orgName := plugin.Namespace
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: orgName}, org); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, fmt.Errorf("organization %s not found", orgName)
+		}
+		return nil, fmt.Errorf("failed to get organization %s: %w", orgName, err)
+	}
+
+	if org.Spec.ConfigMapRef == "" {
+		return nil, nil
+	}
+
+	configMap := &corev1.ConfigMap{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: org.Spec.ConfigMapRef, Namespace: orgName}, configMap); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, fmt.Errorf("organization ConfigMap %s not found in namespace %s", org.Spec.ConfigMapRef, orgName)
+		}
+		return nil, fmt.Errorf("failed to get organization ConfigMap %s: %w", org.Spec.ConfigMapRef, err)
+	}
+
+	registryMirrorData, exists := configMap.Data[registryMirrorConfigKey]
+	if !exists {
+		return nil, nil
+	}
+
+	var mirrorConfig RegistryMirrorConfig
+	if err := yaml.Unmarshal([]byte(registryMirrorData), &mirrorConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse registry mirror configuration: %w", err)
+	}
+
+	if err := validateRegistryMirrorConfig(&mirrorConfig); err != nil {
+		return nil, fmt.Errorf("invalid registry mirror configuration: %w", err)
+	}
+
+	return &mirrorConfig, nil
+}
+
+// validateRegistryMirrorConfig validates the registry mirror configuration.
+func validateRegistryMirrorConfig(config *RegistryMirrorConfig) error {
+	if len(config.RegistryMirrors) == 0 {
+		return errors.New("registryMirrors cannot be empty")
+	}
+
+	for registry, mirror := range config.RegistryMirrors {
+		if registry == "" {
+			return errors.New("registry name cannot be empty")
+		}
+		if mirror.BaseDomain == "" {
+			return fmt.Errorf("basedomain cannot be empty for registry %s", registry)
+		}
+		if mirror.SubPath == "" {
+			return fmt.Errorf("subPath cannot be empty for registry %s", registry)
+		}
+	}
+
+	return nil
+}
+
+// CreateRegistryMirrorPostRenderer creates a Kustomize PostRenderer for registry mirroring.
+// It transforms container images to use mirror registries based on the provided configuration.
+func CreateRegistryMirrorPostRenderer(mirrorConfig *RegistryMirrorConfig) *helmv2.PostRenderer {
+	if mirrorConfig == nil || len(mirrorConfig.RegistryMirrors) == 0 {
+		return nil
+	}
+
+	var images []kustomize.Image
+	for originalRegistry, mirror := range mirrorConfig.RegistryMirrors {
+		newName := fmt.Sprintf("%s/%s", mirror.BaseDomain, mirror.SubPath)
+
+		images = append(images, kustomize.Image{
+			Name:    originalRegistry,
+			NewName: newName,
+		})
+	}
+
+	return &helmv2.PostRenderer{
+		Kustomize: &helmv2.Kustomize{
+			Images: images,
+		},
```

Comment:
This oversimplifies the issue. Actually we need to render the helm chart once in the controller and retrieve all of the container image references from the manifests e.g. `image: <some-image-ref>`
Then for each of these references create an entry of a PostRenderer where the Name is the container image name as found in the rendered manifest and the NewName is where we replace the registry string with the mirror 
e.g. https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/images/